File: dataset_reduzido/avatar/Python/Code/codeforces_445_A.py
n , m = map ( int , input ( ).split ( ) )
l = [ ]
for i in range ( n ) :
    s = input ( )
    s = list ( s )
    for j in range ( m ) :
        if s [ j ] == '.' :
            if ( i + j ) & 1 : s [ j ] = 'W'
            else : s [ j ] = 'B'
    l.append ( s )
for c in l : print ( "".join ( c ) )

File: output/Qwen/avatar/Python/Java/codeforces_445_A.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_445_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);
        
        List<String> l = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String s = scanner.nextLine();
            char[] chars = s.toCharArray();
            for (int j = 0; j < m; j++) {
                if (chars[j] == '.') {
                    if ((i + j) % 2 == 1) {
                        chars[j] = 'W';
                    } else {
                        chars[j] = 'B';
                    }
                }
            }
            l.add(new String(chars));
        }
        
        for (String c : l) {
            System.out.println(c);
        }

        scanner.close();
    }
}